lib/irb/source_finder.rb

```diff
@@ -26,10 +26,18 @@ def find_source(signature)
       when /\A(?<owner>[A-Z]\w*(::[A-Z]\w*)*)#(?<method>[^ :.]+)\z/ # Class#method
         owner = eval(Regexp.last_match[:owner], context_binding)
         method = Regexp.last_match[:method]
-        if owner.respond_to?(:instance_method)
-          methods = owner.instance_methods + owner.private_instance_methods
-          file, line = owner.instance_method(method).source_location if methods.include?(method.to_sym)
+        return unless owner.respond_to?(:instance_method)
+        if s_count
+          super_owner = owner
+          s_count.times  do |s| 
+            if super_owner.superclass
+              super_owner = super_owner.superclass
+            end
+          end
+          owner = super_owner
         end
+        methods = owner.instance_methods + owner.private_instance_methods
+        file, line = owner.instance_method(method).source_location if methods.include?(method.to_sym)
       when /\A((?<receiver>.+)(\.|::))?(?<method>[^ :.]+)\z/ # method, receiver.method, receiver::method
         receiver = eval(Regexp.last_match[:receiver] || 'self', context_binding)
```

Comment:
If we can use the above approach, we can support these signatures too.

Reply:
Just so I'm understanding this correctly.  We would be combining the two different case statements here when I implement the above approach? Or would I be adding this `target_method` logic into each one?

```ruby
 when /\A(?<owner>[A-Z]\w*(::[A-Z]\w*)*)#(?<method>[^ :.]+)\z/ # Class#method

 when /\A((?<receiver>.+)(\.|::))?(?<method>[^ :.]+)\z/ 
```

Reply:
Not combining the case statements. I imagine you can extract the super lookup logic into a private method and use it in both cases instead.

Reply:
Got it

Reply:
Implemented this and refactored `find_source` a bit as well.  Lmk what you think!